server/spec/services/cloud/websocket_client_spec.rb

```diff
@@ -327,5 +327,67 @@
         end
       end
     end
+
+    describe '#resolve_users' do
+      let(:grid) do
+        grid = Grid.create!(name: 'test')
+        grid.users << john
+        grid
+      end
+
+      let(:master_admin_role) do
+        Role.create!(name: 'master_admin', description: 'Master admin')
+      end
+
+      let(:john) do
+        john = User.create(email: 'john.doe@example.org', external_id: '12345')
+        john.roles << master_admin_role
+        john
+      end
+
+      let(:jane) do
+        jane = User.create(email: 'jane.doe@example.org', external_id: '67890')
+        grid.users << jane
+        jane
+      end
+
+      context 'when passing grid_id' do
+        before(:each) do
+          grid
+          john
+          jane
+        end
+
+        context 'when grid is found' do
+          it 'returns master admins and grid users' do
+            users = subject.resolve_users(grid.name)
+            expect(users).to eq([john.external_id, jane.external_id])
+          end
+        end
+
+        context 'when grid is not found' do
+          it 'returns master admins' do
+            grid.destroy
+            users = subject.resolve_users(grid.name)
+            expect(users).to eq([john.external_id])
+          end
+        end
+      end
+
+      context 'when grid_id nil' do
+        before(:each) do
+          grid
+          john
+          jane
+        end
+
+
+        it 'returns master admins' do
+          grid.destroy
```

Comment:
Does this even need the `before` and `grid.destroy` if we're passing `nil`?

Reply:
nope, just some copy paste leftovers. Removed